evadb/executor/executor_utils.py

```diff
@@ -178,7 +178,13 @@ def handle_vector_store_params(
     elif vector_store_type == VectorStoreType.CHROMADB:
         return {"index_path": str(Path(index_path).parent)}
     elif vector_store_type == VectorStoreType.PINECONE:
-        return {}
+        # add the required API_KEYS
+        return {
+            "PINECONE_API_KEY": catalog().get_configuration_catalog_value(
```

Comment:
Is the reason we need to pass a function instead of a catalog instance because of ray?